From 422ec89f6c1fe2f65f1963d97d15b3d94a8c2de5 Mon Sep 17 00:00:00 2001
From: Biswapriyo Nath <nathbappai@gmail.com>
Date: Fri, 11 Jun 2021 19:20:19 +0530
Subject: [PATCH] win32/GNUmakefile: Get CPU arch from target triplet

MinGW-w64 target triplet is different in GCC and Clang/llvm
---
 win32/GNUmakefile | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/win32/GNUmakefile b/win32/GNUmakefile
index ef6042e9e8e..40a000ebc99 100644
--- a/win32/GNUmakefile
+++ b/win32/GNUmakefile
@@ -408,12 +408,16 @@ CCHOME		:= $(VCINSTALLDIR)
 endif
 endif
 
+# Check processor architecture from target triplet
+# possible values:
+# gcc: i686-w64-mingw32, x86_64-w64-mingw32
+# clang: i686-w64-windows-gnu, x86_64-w64-windows-gnu
 ifeq ($(CCTYPE),GCC)
-ifeq ($(GCCTARGET),x86_64-w64-mingw32)
+ifeq (x86_64, $(findstring x86_64, $(GCCTARGET)))
 WIN64 := define
 PROCESSOR_ARCHITECTURE := x64
 endif
-ifeq ($(GCCTARGET),i686-w64-mingw32)
+ifeq (i686, $(findstring i686, $(GCCTARGET)))
 WIN64 := undef
 PROCESSOR_ARCHITECTURE := x86
 endif
